search_agg_terms.go

```diff
@@ -215,3 +312,16 @@ func (t *TermBucketResult) GetAggregationResultSource(name string) ([]byte, bool
 	subAggSource, exists := t.SubAggregationResults[name]
 	return subAggSource, exists
 }
+
+// Keys implemented for [opensearchtools.AggregationResultSet] to return the list of aggregation result keys
+func (t *TermBucketResult) Keys() []string {
+	keys := make([]string, len(t.SubAggregationResults))
```

Comment:
This snippet seems to be identical for all the various aggregation result types.  Could refactor to a shared `subAggregationKeys` function.
```
func subAggregationKeys[K comparable, V any](m map[K]V) []K {
   // exact same code, but ranging over m
}
```
Alternatively, could use [golang.org/x/exp/maps.Keys()](https://pkg.go.dev/golang.org/x/exp/maps#Keys)

Reply:
Didn't know about the `map.Keys()` function, so I can definitely use that.